definitions/src/attributes.rs

```diff
@@ -1,33 +1,32 @@
 use crate::Pointer;
+use num_derive::FromPrimitive;
 
 /// Governs behaviour of EnvironmentAttribute
+#[allow(non_camel_case_types)]
 #[repr(i32)]
-#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+#[derive(Debug, PartialEq, Eq, Clone, Copy, FromPrimitive)]
 pub enum EnvironmentAttribute {
-    OdbcVersion = 200,
-    ConnectionPooling = 201,
-    CpMatch = 202,
-    // This attribute was commented out because there is no mention of it in the ODBC specification
-    // nor does this attribute exist in unixODBC or iODBC implementations. This attribute exists in
-    // Microsoft implementation only and it's usage is unclear.
-    // For private driver manager
-    // SQL_ATTR_APPLICATION_KEY = 203,
-    OutputNts = 10001,
+    SQL_ATTR_ODBC_VERSION = 200,
+    SQL_ATTR_CONNECTION_POOLING = 201,
+    SQL_ATTR_CP_MATCH = 202,
+    SQL_ATTR_OUTPUT_NTS = 10001,
+    SQL_ATTR_DRIVER_UNICODE_TYPE = 1065,
```

Comment:
@terakilobyte 
This is an iODBC specific attribute. See [here](https://github.com/openlink/iODBC/blob/a928c0f37a373dd71c55f5df0ece27b4a05d6de2/include/iodbcext.h#L216)
The DM uses this to check what encoding we are using. Technically this is because iODBC claims that they can do encoding conversion between the app and the driver if needed. In practice, we had no luck so far getting it working correctly I think when we tried to report UTF-16 and that's why we added the feature = "utf32" in cstr.

We should keep it though. The logic is handled correctly in the driver. We might want/be able to use UTF-16 with iODBC one day this way and avoid extra encoding conversion.

I agree that it's missing a comment or two which would explain what the attribute it and what it is used for.
@jchemburkar Would you mind adding a comment?
The section "Unicode Application with a Unicode Driver" on [this page](https://www.progress.com/tutorials/odbc/unicode) describe the idea fairly well, even though this is for the Progress DM. I looked at the iODBC code quite a lot when we tried to get it working, and the feature is the same.

Reply:
yes, I can add a comment to this effect!